Dockerfile

```diff
@@ -1,8 +1,8 @@
-FROM golang:1.21.1 as builder
+FROM golang:1.22.1 as builder
 WORKDIR /build
 COPY go.mod go.sum ./
+COPY *.go ./
 RUN go mod download
-COPY ./ ./
```

Comment:
we want to keep this line because it copies all files recursively. `COPY *.go ./` will only copy the *.go files in the current directory, but we also need all the go files in `pkg/`

Reply:
Ok, makes sense, but I have two small concerns with `COPY . .` :
 - It hoists the default config.yaml file into the build WORKDIR. With the current default start of ENTRYPOINT ["./scratchdata", "config.yaml"], that (I think) could by default point scratchdata at the default config.yaml that got baked into the image, and they would not receive the `INF main.go:31 > No config file specified, using local default values` info message to hint to them that they are running without a config file. 
   - Possibly we should change the entrypoint to not specify a config-file by default, relying on the user to add that as a (docker) runtime argument.
   - Possibly we should a .dockerignore file that ignores config.yaml so the config defaults (or sensitive information like the `crypto: jwt_private_key:` that's in there now doesn't just get baked directly into the docker image.)
 - If a user accidentally creates a data directory in the code repo in the course of testing (as is the current default behavior of `config.yaml`), it would get baked into the image. 
   - Perhaps I/we should add a .dockerignore file that ignores the `./data` folder.